packages/expo-device/android/src/main/java/expo/modules/device/DeviceModule.kt

```diff
@@ -0,0 +1,209 @@
+package expo.modules.device
+
+import expo.modules.core.ExportedModule
+import expo.modules.core.interfaces.RegistryLifecycleListener
+import expo.modules.core.ModuleRegistry
+import expo.modules.core.Promise
+import expo.modules.core.interfaces.ActivityProvider
+import expo.modules.core.interfaces.ExpoMethod
+import expo.modules.core.ModuleRegistryDelegate
+
+import com.facebook.device.yearclass.YearClass
+
+import android.app.Activity
+import android.app.ActivityManager
+import android.app.UiModeManager
+import android.content.Context
+import android.content.res.Configuration
+import android.os.Build
+import android.os.SystemClock
+import android.provider.Settings
+import android.view.WindowManager
+import android.util.DisplayMetrics
+
+import java.io.File
+import kotlin.math.pow
+import kotlin.math.sqrt
+
+private const val NAME = "ExpoDevice"
+
+class DeviceModule(private val mContext: Context) : ExportedModule(mContext), RegistryLifecycleListener {
+  private val moduleRegistryDelegate: ModuleRegistryDelegate = ModuleRegistryDelegate()
+  private val mActivityProvider: ActivityProvider? by moduleRegistry()
```

Comment:
It would be nice if this could be made non-nullable to help avoid needing to use `!!` and `?.` at the callsites. I'm not 100% sure it's always non-null after `onCreate` is called, but I think it probably is since places like `StoreReviewModule` use lateinit and a non-null type. 
```suggestion
  private val mActivityProvider: ActivityProvider by moduleRegistry()
```